includes/class-activitypub.php

```diff
@@ -332,18 +332,15 @@ public static function pre_get_avatar_data( $args, $id_or_email ) {
 			return $args;
 		}
 
+		/**
+		 * Filter allowed comment types for avatars.
+		 *
+		 * @param array $allowed_comment_types Array of allowed comment types.
+		 */
 		$allowed_comment_types = \apply_filters( 'get_avatar_comment_types', array( 'comment' ) );
-		if (
-			! empty( $id_or_email->comment_type ) &&
-			! \in_array(
-				$id_or_email->comment_type,
-				(array) $allowed_comment_types,
-				true
-			)
-		) {
+		if ( ! \in_array( $id_or_email->comment_type ?: 'comment', $allowed_comment_types, true ) ) { // phpcs:ignore Universal.Operators.DisallowShortTernary
 			$args['url'] = false;
```

Comment:
@pfefferle I think I'm missing historic context, hoping for you to be able to fill in my blanks.

My understanding is that setting the `url` will [short-circuit](https://github.com/WordPress/wordpress-develop/blob/a49176e653d53c273ae7a94c0791da2befabb439/src/wp-includes/link-template.php#L4471-L4473) `get_avatar_data()` for all cases where the comment type is not part of the `$allowed_comment_types` array, so things like pings and trackbacks and such. Is that on purpose?

Reply:
but I have no strong feelings about that... happy to remove it!